app/Metadata/Json/ExternalRequestFunctions.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2017-2018 Tobias Reich
+ * Copyright (c) 2018-2025 LycheeOrg.
+ */
+
+namespace App\Metadata\Json;
+
+use App\Contracts\ExternalRequest;
+use App\Exceptions\Internal\RequestFailedException;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use function Safe\file_get_contents;
+use function Safe\ini_get;
+
+class ExternalRequestFunctions implements ExternalRequest
+{
+	protected ?string $data = null;
+
+	public function __construct(
+		private string $url,
+		private int $ttl,
+	) {
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public function clear_cache(): void
+	{
+		Cache::forget($this->url);
+		Cache::forget($this->url . '_age');
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_age_text(): string
+	{
+		$age = Cache::get($this->url . '_age');
+		if (!$age instanceof \DateTimeInterface) {
+			// @codeCoverageIgnoreStart
+			return 'unknown';
+			// @codeCoverageIgnoreEnd
+		}
+		try {
+			$text = match (0) {
+				(int) now()->diffInMinutes($age) => (int) -now()->diffInSeconds($age) . ' seconds',
+				// @codeCoverageIgnoreStart
+				(int) now()->diffInHours($age) => (int) -now()->diffInMinutes($age) . ' minutes',
+				(int) now()->diffInDays($age) => (int) -now()->diffInHours($age) . ' hours',
+				(int) now()->diffInWeeks($age) => (int) -now()->diffInDays($age) . ' days',
+				(int) now()->diffInMonths($age) => (int) -now()->diffInWeeks($age) . ' weeks',
+				(int) now()->diffInYears($age) => (int) -now()->diffInMonths($age) . ' months',
+				default => now()->diffInYears($age) . ' years',
+				// @codeCoverageIgnoreEnd
+			};
+
+			return $text . ' ago';
+			// @codeCoverageIgnoreStart
+		} catch (\Throwable) {
+			return 'unknown';
```

Comment:
if we return unknown , that means that we have to handle this unknow case somehwere else

like ` if (get_age_text() == unknown) { //do shizzle}` either throw an exception or return null

Reply:
Nah, this is directly returned to the front-end.
Ideally this function should actually be localized...